Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/TaskRecyclerViewAdapter.java

```diff
@@ -0,0 +1,124 @@
+package pg.autyzm.friendly_plans;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import database.entities.TaskTemplate;
+
+
+/**
+ * Created by Mateusz on 2017-03-08.
+ */
+
+public class TaskRecyclerViewAdapter extends RecyclerView.Adapter<TaskRecyclerViewAdapter.TaskListViewHolder> {
+
+
+    private List<TaskTemplate> taskItemList;
+    private TaskItemClickListener taskItemClickListener;
+    private int ICON_PLACEHOLDER_ID = R.drawable.ic_placeholder;
+
+    @Override
+    public TaskListViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_task, parent, false);
+        return new TaskListViewHolder(view, taskItemClickListener);
+    }
+
+    @Override
+    public void onBindViewHolder(TaskListViewHolder holder, int position) {
+
+        if (taskItemList != null || taskItemList.size() != 0) {
```

Comment:
Is it necessary check? I think it should never happen - onBindViewHolder won't be called when taskItemList is null or have no elements, so I think we can remove this check.

PS: Checking it this way may cause an exception (not in our case but in general). For example, if you'd call onBindViewHolder when taskItemList is null the condition would be evaluated in such way:
1) Check left part of condition -> false
2) Check right part of condition (because there is OR between them) -> null pointer exception

I guess you'd like to use && operator (then if left part of condition is false rest of it won't be evaluated) :)